### DISCUSSION TITLE
Problems in Gradio, and Jupyter

### DISCUSSION THREAD

**Discussion by rajavishah** [bilayer-containers/bilayers#25] [Q&A]
I just realised that `gradio` and `jupyter` takes the default value for the things whose input isn't entered by the user.

Which is good for the following scenarios -
1. `Chan` : where I believe you should have argument. like GRAY if nothing is selected

But it’s not good for the following scenarios -
1. `PreTrained Model` & `Add Model `: So, even if user gives a file for Add Model, it still takes the default value present in PreTrained Model, which would lead giving incorrect outputs
2. `Chan2` is optional, though it always takes None, which is not the desired output for Image Analysts, right?

But if I handle all the exception cases explicitly, it would hamper the resilience of the `generate.py` file. Any suggestions on how to handle these both cases?

**Comment by bethac07** [comment]
I suspect the correct answer here for the first case is that the spec should have default setting values for each setting. @gnodar01 agree?

**Reply by rajavishah** [reply]
No, we actually have default setting. So, if nothing is present it takes the default value.

For eg. if in checkbox the default is `False` for everything, hence it takes default value and doesnot append the tags which have False value. Similarly, `Chan` has `GRAY` as default value

**Comment by bethac07** [comment]
For the second case, I suspect there's no way around some dependency mapping; "Dependency on other inputs" is in the Google Sheets spec. I suspect least-worst thing is something like 

```
"ignore_if_not_set": ["some_other_setting_1"]
"ignore_if_set": ["some_other_setting_2"]
```

**Reply by rajavishah** [reply]
Yeah, that seems fair!

**Reply by rajavishah** [reply]
@gnodar01 and I discussed, that this approach is going into the complexity at code level that for each input we should consider validating if it has some dependency or not. Moreover, iterating over multiple inputs and checking the Priority* too is adding the complexity layer. Instead, an easy way out would be adding one more option to `RadioButton` or `Dropdown` which says `ignore`. If the user, opts for `ignore` option, we wouldn't consider that flag at all. 

And as posted in this [discussion](https://github.com/orgs/bilayer-containers/discussions/20), we could handle rigorous validations in future. 

What we could handle as part of this version is : 
a. When the `gradio` throws error, we would display that to the screen and ask the user to put the screenshot of that dialogue box as a Github issue and we could look into that!
b. When `jupyter` throws error, it would be displayed anyways on the screen. Hence, just a markdown with instruction that if you see any error just take a screenshot and put that as Github issue.

Any thoughts around this approach @bethac07 ?

*Priority = (For eg. if user-input values for both `pretrained_model` and `add_model` are present then consider `add_model`)

**Comment by rajavishah** [comment]
Also, for future refer https://github.com/orgs/bilayer-containers/discussions/20 - what more can be implemented
